### PR TITLE
Avoid exceptions for control flow

### LINKED ISSUE #40427: "No implementation found for method TextInput.setEditableSizeAndTransform on channel flutter/textinput"
Hi,

I just hit this exception when I have `Break on all exceptions`enabled in VS code and I click on a TextFormField on my Emulator.
Is this as designed?

image
```
[√] Flutter (Channel master, v1.10.2-pre.4, on Microsoft Windows [Version 10.0.18362.295], locale de-DE)
    • Flutter version 1.10.2-pre.4 at C:\Entwicklung\flutter
    • Framework revision 5a3a46ada2 (6 days ago), 2019-09-07 18:37:41 -0400
    • Engine revision c9ea4dba8d
    • Dart version 2.5.0 (build 2.5.0-dev.4.0 ec7ec4ecf7)

 
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at C:\Users\escam\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[√] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] IntelliJ IDEA Community Edition (version 2018.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.2.5
    • Flutter plugin version 29.1.3
    • Dart plugin version 182.4892.25

[√] VS Code, 64-bit edition (version 1.38.0)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.4.1

[√] Connected device (1 available)
    • Android SDK built for x86 64 • emulator-5554 • android-x64 • Android 7.1.1 (API 25) (emulator)
```

### PR DESCRIPTION
**Pull request by dnfield**
## Description

This ensures that we don't use exceptions for control flow in `OptionalMethodChannel`s, which can cause unpleasant debugging experiences for users trying to catch all exceptions.

We should still look at whether we're doing work we shouldn't be by calling this at all separately - filed https://github.com/flutter/flutter/issues/46896 to track that separately.

## Related Issues

Fixes #40427
Original attempt at https://github.com/flutter/flutter/pull/46843

## Tests

I added the following tests:

A test covering `OptionalMethodChannel.invokeMethod`.  Unfortunately, I cannot find a good way to assert that we're no longer using exceptions for control flow (even using my own Zone, I only get the exception if it was not handled, which the old code did).

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/services/platform_channel.dart [hunk 1]
```diff
@@ -309,17 +325,8 @@ class MethodChannel {
...
-      name,
-      codec.encodeMethodCall(MethodCall(method, arguments)),
-    );
-    if (result == null) {
-      throw MissingPluginException('No implementation found for method $method on channel $name');
-    }
-    final T typedResult = codec.decodeEnvelope(result) as T;
-    return typedResult;
+  Future<T> invokeMethod<T>(String method, [ dynamic arguments ]) {
+    return _invokeMethod(method, missingOk: false, arguments: arguments);
```
Comment (mdebbar) [inline packages/flutter/lib/src/services/platform_channel.dart]: You may want to pass `T` as well (i.e. `return _invokeMethod<T>(...)`)?

**Review by mdebbar — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/services/platform_channel.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/lib/src/services/platform_channel.dart]: Done

**Review by nturgut — APPROVED** [review]
Thanks for solving this issue!

**Review by yjbanov — APPROVED** [review]

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by chickenblood** [comment]
Thanks for fixing this Dan!

**Pull request merged**